packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -129,6 +129,21 @@ export default createRule({
           caseTypes.add(getNodeType(switchCase.test));
         }
 
+        const isFiniteType = unionTypes.every(isLiteralType);
+        const isSymbolType = unionTypes.some(type =>
+          isTypeFlagSet(type, ts.TypeFlags.UniqueESSymbol),
+        );
+        const isStringLiteralOrBoolean = unionTypes.some(type =>
+          isTypeFlagSet(
+            type,
+            ts.TypeFlags.BooleanLiteral || ts.TypeFlags.StringLiteral,
```

Comment:
[Bug] `||` is the _logical_ or operator. I'm guessing you meant `|` as the _bitwise_ or.

But, this feels weird to me. It's generally not ideal for us to be hardcoding to specific primitives. What about bigint literals, number literals, etc.? This code should handle them too. `ts.TypeFlags.Literal` should be what you want to use.

Could you also add some test cases with bigint literals (`1n`) and template literal strings (<code>`a`</code>)?